tests/Unit/Service/CommunicationServiceTest.php

```diff
@@ -168,6 +182,18 @@ public function testSendAndRetryCurl()
             $this->assertEquals('cURL Executed!', $response);
         }
 
+        public function testCurlRequestWillFailWithHttpError500()
+        {
+            self::$forceHttpStatus = 500;
+            $xml = '<?xml version="1.0" encoding="UTF-8"?><note><to>Foo</to></note>';
+
+            $this->expectException(CurlException::class);
+            $this->expectExceptionMessage('There server answered with an invalid status code.');
+
+            $service = new CommunicationService();
+            $service->send($xml, 0, 0, 6, 1);
+        }
+
         public static function incrementCurlExcecutions()
```

Comment:
Could yozu please also add the tests for the other cases `x < 200` and `300 < x >400` for a better coverage.

Reply:
Hi @rommelfreddy, Any news about this PR? Let us get this through and merge :)